### ISSUE TITLE
CR: Add aria-labelledby and alt-text to all Guide Cards on the NCIDS

### ISSUE THREAD

**Issue by dev-rana-publicis** [NCIOCPL/ncids#1112]
### Request Description
On the platform, there is an `aria-labelledby` on the `ul` for the link buttons for each of the Guide Cards and an `id` on the Guide Card Heading (not the row). The request is to update all instances of the Guide Cards on the DS to appropriately handle the accessibility for this component. This will impact: 
- [The Guide Card page on the Documentation Site](https://designsystem-dev.cancer.gov/develop/components/guide-card) for both Default Guide Card and Guide Card with Image and Description
- [The Guide Card Row page on the Documentation Site](https://designsystem-dev.cancer.gov/develop/templates/guide-card-row) for both 2-Guide Card Row and 3-Guide Card Row
- All Guide Card Stories on Storybook starting with [Guide Card Default](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-guide-card-default--default)

### Expected Functionality
- All lists need the `aria-labelledby` to indicate what the group of links are. Therefore, the `aria-labeledby` shall be included on all the components and code previews for the unordered list on Guide Cards on the NCIDS. This impacts all stories and code snippets.
- When the Guide Card has a title, then the `aria-labeledby` shall be set to be the title of the card. This should be done through the use of the `id` as used on prod. This impacts Guide Card Default, Guide Card Default Card Group, Title Aligned Card Group Without Description, Title Aligned Card Group Without Image, and Title Aligned Card Group.
- When a Guide Card does not have a title, it will still need an `aria-label`, but there is no title to link the `id` to. In these cases, the list will have an `aria-label` set to "<Add accessible label here.>" This impacts Guide with Image and Description without Image, Guide Card with Image and Description without Description, and Single Guide Card with Image and Description without Heading.
- When the Guide Card has an image, then the image shall have `alt-text`. The `alt-text` field should be set to the empty string if the image is decorative. This impacts Guide Card with Image and Description without Description, Title Aligned Card Group without Description, Single Guide Card With Image and Description Without Heading, and Title Aligned Card Group.

### Current Functionality
- There is no `aria-labelledby` on the unordered list on Guide Card Cards on the NCIDS
- Images do not have `alt-text`

### Related Tickets
* NCIDS Documentation Site Ticket: #1174 
* Original Story Ticket #614

**Comment by bryanpizzillo** [comment]
This ticket needs to be revisited. From what we can tell, giving a label to a list is fairly useless. It was the guidance at the time when this component was implemented on the platform. Based on some research it may make sense that the guide cards with image and description should be a `<section>`. However, for the 3 guide-card row, it would be odder for each card to be a section. (A standalone "normal" guide card could be a section).

The point is, the list should not be labeled as it does not help anyone find the links more quickly, nor does it seem that links get context.

@andyvanavery31 we should put this on hold.

**Comment by andyvanavery31** [comment]
@bryanpizzillo @olitharp-nci upon review of the note above, the slack thread, and reviewing the components, I agree that labelling the list is redundant and not useful. Additionally, putting the `id` on the `nci_guide-card__heading` is also redundant because the h2 already is evaluated. Therefore, if you agree, my proposal is that we close this ticket, and open a new one on the Digital Platform to change the guide cards on prod to not have the `id` on the heading, and to not have the `aria-labeledby` on the list.
